### PR TITLE
in target_* function, drop groups and warn if input data is grouped

### LINKED ISSUE #114: target_sda() outputs a grouped data frame. Is this intentional?
``` r
library(r2dii.data)
library(r2dii.match)
library(r2dii.analysis)

valid_matches <- match_name(loanbook_demo, ald_demo) %>%
  prioritize()

out <- valid_matches %>%
  target_sda(
    ald = ald_demo,
    co2_intensity_scenario = co2_intensity_scenario_demo
  )

dplyr::is_grouped_df(out)
#> [1] TRUE
```

<sup>Created on 2020-06-24 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

### LINKED ISSUE #117: target_market_share() may need to drop groups
In general I expect a function to preserve groups, but that is impossible if the output excludes the grouping variable.  This example throws an error because the grouping variable in the input `data` is not present in the output.

``` r
library(r2dii.match)
library(r2dii.analysis)
devtools::load_all()
#> Loading r2dii.analysis

grouped_by_column_not_in_output <- fake_matched() %>% 
  dplyr::group_by(.data$id_loan)

target_market_share(
  grouped_by_column_not_in_output,
  fake_ald(),
  fake_scenario(),
  r2dii.data::region_isos_demo
)
#> Error: Must group by variables found in `.data`.
#> * Column `id_loan` is not found.
```

<sup>Created on 2020-06-24 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

For all functions that return columns different than those passed to data, it is hard to promise we'll keep groups. I suggest we drop them with a warning. This will keep our promise simple, and the code simpler than it currently is.

### PR DESCRIPTION
**Pull request by jdhoffa**
Closes #114 
Closes #117

### CODE REVIEW

**Comment by jdhoffa** [comment]
I have elected the more strict option: 
- ungroup input data
- warn when doing so
- output ungrouped data. 

I can see a use-case where we may want to revisit this choice. As a concrete example: a financial supervisor may want to run multiple loanbooks. It would be easy to just add a column title "loanbook_name" and group by this to calculate targets per loanbook. That said, it is a little tricky because there are obviously internal groupings happening etc. so we would have to be clever about which input groupings are allowed (we could warn if any crucial columns are grouped, and ungroup only those columns or something like this). I think this is something we should aim for in version-0.0.2. 

For now, I'd like to prioritize #107 and #113

File: R/target_market_share.R [hunk 1]
```diff
@@ -4,6 +4,8 @@
 #' calculated using the market share approach applied to each relevant climate
 #' production forecast.
 #'
+#' @template ignores-existing-groups
```
Comment (maurolepore) [inline R/target_market_share.R:7]: Well done!

File: R/target_market_share.R [hunk 2]
```diff
@@ -62,9 +64,7 @@ target_market_share <- function(data,
                                 by_company = FALSE) {
   stopifnot(is.data.frame(data))
 
-  old_groups <- dplyr::groups(data)
-  data <- ungroup(data)
-
+  data <- ungroup_and_warn(data, "Ungrouping input data.")
```
Comment (maurolepore) [inline R/target_market_share.R:67]: :+1:

File: R/target_market_share.R [hunk 3]
```diff
@@ -174,7 +174,7 @@ target_market_share <- function(data,
       names_to = "weighted_production_metric",
       values_to = "weighted_production_value"
     ) %>%
-    group_by(!!!old_groups)
+    ungroup()
```
Comment (maurolepore) [inline R/target_market_share.R:177]: :+1:

File: tests/testthat/test-target_market_share.R [hunk 4]
```diff
@@ -16,6 +16,30 @@ test_that("outputs a tibble", {
   expect_is(out, "tbl_df")
 })
 
+test_that("outputs is ungrouped", {
```
Comment (maurolepore) [inline tests/testthat/test-target_market_share.R:19]: :1st_place_medal:

**Comment by maurolepore** [comment]
Looks great. Please review my changes and merge when you are ready.

**Pull request merged**